src/Microsoft.AspNetCore.StaticFiles/ResponseCacheFilter.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.StaticFiles
+{
+    internal class ResponseCacheFilter : IResponseCacheFilter
+    {
+        /// <summary>
+        /// Apply the given cacheProfile to the <see cref="HttpContext"/>
+        /// </summary>
+        /// <param name="context">The current <see cref="HttpContext"/></param>
+        /// <param name="cacheProfile">The profile which contains the cache control settings</param>
+        public void ApplyCacheProfile(HttpContext context, CacheProfile cacheProfile)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            if (cacheProfile == null)
+            {
+                throw new ArgumentNullException(nameof(cacheProfile));
+            }
+
+            var noStore = cacheProfile.NoStore ?? false;
```

Comment:
The main reason for not setting default values in the `CacheProfile` was so the properties could be nullable. That way we can check that `Duration` was explicitly set if `NoStore` is false, rather than just using the default (0) value, and having people accidentally setting a max-age of 0. 

Again, this was following the lead of the MVC filter, so happy to change it as long as we don't mind the difference in behaviour.